### PR TITLE
Refactor rct_research_item, refactor access to rct_research_item

### PR DESCRIPTION
**Pull request by Gymnasiast**
This takes away a lot of the bitshift and bit mask trickery previously used.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
I think we should just perform an import and export on load/save for the research and redo the structure completely.

**Comment by Gymnasiast** [comment]
@duncanspumpkin I am already working on that. But I want to
- fully understand research (RCT2's very own Japanese tea ceremony)
- make sure the changes are easy to follow.

And that is why I did this.

**Comment by duncanspumpkin** [comment]
Yeah i kinda realised that after I posted that this was a good stepping stone. Good work.

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 1]
```diff
@@ -284,12 +284,12 @@ static void research_scenery_sets_setup(){
             continue;
 
         rct_research_item* research = gResearchItems;
-        for (; research->entryIndex != RESEARCHED_ITEMS_END; research++){
+        for (; research->rawValue != RESEARCHED_ITEMS_END; research++){
 
-            if ((research->entryIndex & 0xFFFFFF) != entry_index)
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/EditorInventionsList.cpp]: rawValue

**Pull request merged**